exp/runtime/hooks/api/v1alpha1/util.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"reflect"
+)
+
+const (
+	fieldStatus            = "Status"
+	fieldMessage           = "Message"
+	fieldRetryAfterSeconds = "RetryAfterSeconds"
+)
+
+func valueOf(obj interface{}) (reflect.Value, error) {
+	var v reflect.Value
+	var t reflect.Type
+	if reflect.TypeOf(obj).Kind() == reflect.Ptr {
+		t = reflect.TypeOf(obj).Elem()
+		v = reflect.ValueOf(obj).Elem()
+	} else {
+		t = reflect.TypeOf(obj)
+		v = reflect.ValueOf(obj)
+	}
+	if t.Kind() != reflect.Struct {
+		return v, fmt.Errorf("unexpected type")
+	}
+	return v, nil
+}
+
+func IsFailure(obj interface{}) (bool, error) {
+	status, err := GetStatus(obj)
+	if err != nil {
+		return false, err
+	}
+	return status == ResponseStatusFailure, nil
+}
+
+func GetStatus(obj interface{}) (ResponseStatus, error) {
+	status, err := getString(obj, fieldStatus)
```

Comment:
I would prefer to `return nil, err` if `err != nil` (+ error wrapping eventually)